### PR TITLE
Use common sizing routines for sizing DX Coil SHR

### LINKED ISSUE #7788: Convert Coil:Cooling:DX:Multispeed to use centralized sizing routine
Issue overview
--------------
Code maintenance requires a change to the multispeed DX coil to use a centralized sizing routine. Main changes will be in autosized coil SHR where a test is used to make sure the SHR is valid and results in a valid rated outlet condition (i.e., results fall on the psyc chart).

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
The multispeed DX coil calculated a coil SHR differently than other DX coils. This change moves coil SHR sizing to a centralized sizing routine.

 - Fixes #7788
 - Convert DX Coil SHR sizing to use RequestSizing. Will cause diff's since RequestSizing uses additional test for valid coil SHR given air flow and capacity.
After SHR is sized, RequestSizing also calls ValidateADP to confirm the SHR is valid.

    // check that the autosized SHR corresponds to a valid apperatus dew point (ADP) temperature
    AutosizeDes = ValidateADP(CompType, CompName, RatedInletAirTemp,
                              RatedInletAirHumRat, DataCapacityUsedForSizing,
                              DataFlowUsedForSizing, AutosizeDes, CallingRoutine);

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -7463,103 +7463,75 @@ namespace DXCoils {
                 if (DXCoil(DXCoilNum).MSRatedSHR(Mode) == AutoSize) {
                     IsAutoSize = true;
                 }
-                if ( Mode == DXCoil( DXCoilNum ).NumOfSpeeds ) {
-                    if ( CurSysNum > 0 ) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/DXCoils.cc:7467]: So this CurSysNum check is no longer needed?

File: src/EnergyPlus/DXCoils.cc [hunk 2]
```diff
@@ -7463,103 +7463,75 @@ namespace DXCoils {
...
-                            if ( MSRatedTotCapDesAtMaxSpeed > 0.0 ) {
-                                RatedVolFlowPerRatedTotCap = MSRatedAirVolFlowRateDes / MSRatedTotCapDesAtMaxSpeed;
-                            } else {
-                                RatedVolFlowPerRatedTotCap = 0.0;
-                            }
-                        } else {
-                            RatedVolFlowPerRatedTotCap = DXCoil(DXCoilNum).MSRatedAirVolFlowRate(Mode) / DXCoil(DXCoilNum).MSRatedTotCap(Mode);
-                        }
-                        if (RatedVolFlowPerRatedTotCap > MaxRatedVolFlowPerRatedTotCap(DXCT)) {
-                            MSRatedSHRDes = 0.431 + 6086.0 * MaxRatedVolFlowPerRatedTotCap(DXCT);
```
Comment (Myoldmopar) [inline src/EnergyPlus/DXCoils.cc:7508]: Glad these coefficients are centralized in one area of the code.

File: src/EnergyPlus/DXCoils.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/DXCoils.cc:7467]: No, RequestSizing handles all of that.

Comment (Myoldmopar) [reply src/EnergyPlus/DXCoils.cc:7467]: Excellent.  I'm building and running some tests locally and we'll try to get this merged ASAP.

**Comment by rraustad** [comment]
There is also some other hoopalah going on here where FlowPerCapacity is calculated based on these locals if the SHR is autosized, yet these variables are not always set based on the actual flow rate. Strange, but I didn't change that. Main difference is the call to ValidateADP which adjusts the SHR as needed to remain on the psyc chart.

if Rated SHR is autosized:
    RatedVolFlowPerRatedTotCap = MSRatedAirVolFlowRateDes / MSRatedTotCapDesAtMaxSpeed;
else
    RatedVolFlowPerRatedTotCap = DXCoil(DXCoilNum).MSRatedAirVolFlowRate(Mode) / DXCoil(DXCoilNum).MSRatedTotCap(Mode);

**Comment by rraustad** [comment]
Coil SHR diffs as expected. I would, in the future, change this again to get rid of the hoopalah, which will change the SHR again for multispeed colils.

**Comment by Myoldmopar** [comment]
This looks good, CI is as expected.  I'm going to call it complete so we can move forward.  Thanks for the quick turnaround on this @rraustad

**Pull request merged**